### PR TITLE
Setup the base for tauri, vite and vuejs

### PR DESCRIPTION
**Pull request by Leuchak**
This pull request marks the first step for a significant shift in our technology stack, introducing the foundation of Tauri, Vue.js, and Vite to our project. Here's a high-level overview of the major changes and the rationale behind them:

**1. Adoption of Tauri (Replacing Electron):**

We're making a transition from Electron to Tauri for building cross-platform desktop applications. Tauri offers a leaner bundle size and improved performance compared to Electron, making it a compelling choice for modern desktop app development.

**2. Vue.js Adoption (Replacing React):**

We've decided to embrace Vue.js as our front-end framework, supplanting React. Vue.js is renowned for its simplicity and gentle learning curve, and it empowers us to create dynamic user interfaces efficiently.

**3. Vite Integration (Replacing Snowpack):**

We're integrating Vite as our build and development tool, replacing Snowpack. Vite's rapid compilation speed and native ES6 module support are well-suited for modern web applications.

These changes represent a strategic decision to align our technology stack with current industry trends and best practices. While these shifts may involve some learning curves, they pave the way for a more efficient and performant development process.

We encourage thorough review and testing of these modifications to ensure a smooth transition and to identify any potential issues. Your feedback and insights are highly valuable as we embark on this exciting journey with Tauri, Vue.js, and Vite.

**Tasks to be completed:**

- [x] Integration of Tauri
- [x] Adoption of Vue.js
- [x] Integration of Vite
- [x] Integration of Vitest
- [x] Updated documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,13 +58,13 @@ npm i
 - Bundle the typescript in src/main and src/renderer in `/build` and use those to build the electron app installer. The final executable will be found in `/dist`
```
Comment (mmmhsalois) [inline README.md:58]: Mentioning Electron, has the doc been updated?

Comment (mmmhsalois) [reply README.md:58]: Also, it is really slow on WSL.

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    // Resolver
+    resolve: {
+      // https://vitejs.dev/config/shared-options.html#resolve-alias
+      alias: {
+        '@': fileURLToPath(new URL('./src', import.meta.url)),
+      },
+      extensions: ['.js', '.json', '.jsx', '.mjs', '.ts', '.tsx', '.vue'],
+    },
+    // 3. to make use of `TAURI_DEBUG` and other env variables
```
Comment (mmmhsalois) [inline vite.config.ts:57]: Where are the 1 and 2???

**Review by mmmhsalois — CHANGES_REQUESTED** [review]

File: vite.config.ts (hunk 2 above)
Comment (Leuchak) [reply vite.config.ts:57]: oops

File: README.md (hunk 1 above)
Comment (Leuchak) [reply README.md:58]: I've added a note to tell Windows users not to use WSL

File: vite.config.ts (hunk 2 above)
Comment (Leuchak) [reply vite.config.ts:57]: removed

**Review by mmmhsalois — APPROVED** [review]

**Pull request merged**